src/components/DataTable/DataTable.tsx

```diff
@@ -52,18 +57,44 @@ const extractIconFromCell = (cellValue: unknown): React.ReactElement | undefined
 export const MoonstoneTable = <T extends Record<string, unknown>>({
     data,
     columns,
-    isStructured = false
+    isStructured = false,
+    enableSelection = false,
+    onChangeSelection,
+    isSortable = false
 }: DataTableProps<T>) => {
     const [expanded, setExpanded] = useState<ExpandedState>({});
+    const [rowSelection, setRowSelection] = useState<RowSelectionState>({});
+    const [sortState, setSortState] = useState<{columnId: string; direction: 'ascending' | 'descending'} | undefined>(undefined);
+
+    useEffect(() => {
+        onChangeSelection?.(rowSelection);
+    }, [rowSelection, onChangeSelection]);
+
+    const handleHeaderClick = (columnId: string) => {
+        setSortState(prevState => {
+            if (prevState?.columnId === columnId) {
+                if (prevState.direction === 'ascending') {
+                    return {columnId, direction: 'descending'};
+                }
+                return undefined;
+            }
+            return {columnId, direction: 'ascending'};
+        });
+    };
```

Comment:
If the component doesn't manage sorting itself, you shouldn't manage the direction or the sorted column, as they aren't reflected in the table data.

So we don't need a state. We just display the correctly sorted column in the right direction. You have to provide an `onClick` event on the `TableHead` so the consumer can manage it for us.